### PR TITLE
Enhancement/1848 permissions error notification

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1848 

## Relevant technical choices

- > Update the ErrorNotifiction component to return the new AuthError when applicable.
  
  Not sure if it's needed since I have already added `AuthError` via filter hook mentioned above in the IB.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by felixarntz — COMMENTED** [review]
Mostly LGTM, just some minor feedback.

File: assets/js/googlesitekit/datastore/user/authentication.js [hunk 1]
```diff
@@ -48,10 +48,59 @@ const fetchGetAuthenticationStore = createFetchStore( {
 	},
 } );
 
+// Actions
+const SET_AUTH_ERROR = 'SET_AUTH_ERROR';
+
 const baseInitialState = {
 	authentication: undefined,
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/authentication.js:56]: Updated.

File: assets/js/googlesitekit/datastore/user/authentication.js [hunk 2]
```diff
@@ -48,10 +48,59 @@ const fetchGetAuthenticationStore = createFetchStore( {
...
+	 * Clears the authentication error, if one was previously set.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Redux-style action.
+	 */
+	clearAuthError() {
+		return {
+			payload: {},
+			type: SET_AUTH_ERROR,
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/authentication.js]: Updated.

File: assets/js/googlesitekit/datastore/user/authentication.js [hunk 3]
```diff
@@ -184,12 +233,27 @@ const baseSelectors = {
...
 	 */
 	getDisconnectedReason: createGetAuthenticationSelector( 'disconnectedReason' ),
+
+	/**
+	 * Gets the authentication error.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Authentication error object if available, otherwise undefined.
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/googlesitekit/datastore/user/authentication.js]: Updated.

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**